test/optim/test_initializers.py

```diff
@@ -270,6 +273,115 @@ def test_gen_batch_initial_conditions_warning(self):
                     )
                 )
 
+    def test_gen_batch_initial_conditions_transform_horizontal_constraint(self):
+        for dtype in (torch.float, torch.double):
+            constraint = (
+                torch.tensor([0, 1], dtype=torch.int64, device=self.device),
+                torch.tensor([-1, -1]).to(dtype=dtype, device=self.device),
+                -1.0,
+            )
+            constraints = transform_horizontal_constraint(
+                constraint=constraint, d=3, q=3
+            )
+            self.assertEqual(len(constraints), 3)
+            self.assertAllClose(
+                constraints[0][0],
+                torch.tensor([0, 1], dtype=torch.int64, device=self.device),
+            )
+            self.assertAllClose(
+                constraints[1][0],
+                torch.tensor([3, 4], dtype=torch.int64, device=self.device),
+            )
+            self.assertAllClose(
+                constraints[2][0],
+                torch.tensor([6, 7], dtype=torch.int64, device=self.device),
+            )
+            for constraint in constraints:
+                self.assertAllClose(
+                    torch.tensor([-1, -1]).to(dtype=dtype), constraint[1]
+                )
+                self.assertEqual(constraint[2], -1.0)
+            # test failure on invalid d
+            constraint = (
+                torch.tensor([[0, 3]]),
+                torch.tensor([-1.0, -1.0], dtype=dtype),
+                0,
+            )
+            with self.assertRaises(ValueError):
```

Comment:
suggest using `assertRaisesRegex` here and check for the error message (once you've added that)

Reply:
DOne.